wrappers/src/sync_session_cs.hpp

```diff
@@ -23,12 +23,14 @@
 #include <realm/sync/config.hpp>
 
 using ProgressCallbackT = void(void* state, uint64_t transferred_bytes, uint64_t transferrable_bytes);
+using NotifyBeforeClientResetCallbackT = bool(SharedRealm& before_frozen, void* managed_client_reset_handler_handle);
+using NotifyAfterClientResetCallbackT = bool(SharedRealm& before_frozen, SharedRealm& after, void* managed_client_reset_handler_handle);
```

Comment:
This has a slightly different signature from the migration callback where we use pointers instead of references. I don't know how important it is or if's a problem, but is strange. cc @fealebenpae